docs/proposal/multi-target.md

```diff
@@ -0,0 +1,119 @@
+    
+2. https://github.com/kubernetes-incubator/external-dns/pull/261 - attempt to rework `plan` to make it work correctly with multiple targets. 
+    
+    *what it does* : attempts to fix issues with `plan` described in `Current Behaviour` section above. Included tests reveal the current problem with `plan`
+    
+    *action*: rebase on master and make necessary changes to satisfy requirements listed in this document 
+    
+3. https://github.com/kubernetes-incubator/external-dns/pull/326 - attempt to add multiple target support. 
+    
+    *what it does*: for each pair `DNS Name` + `Record Type` it aggregates **all** targets from the cluster and passes them to Provider. It adds basic support
+    for DO, Azura, Cloudflare, AWS, GCP, however those are not tested (?). (DNSSimple and Infoblox providers were not updated)
+    
+    *action*: the `plan` logic will probably needs to be reworked, however the rest concerning support in Providers and extending `Endpoint` struct can be reused. 
+    Rebase on master and add missing pieces. Depends on `2`. 
+    
+ Related PRs: https://github.com/kubernetes-incubator/external-dns/pull/331/files,  https://github.com/kubernetes-incubator/external-dns/pull/347/files - aiming at AWS Route53 weighted records.
+These PRs should be considered after common agreement about the way to address multi-target support is achieved. Related discussion:  https://github.com/kubernetes-incubator/external-dns/issues/196
+
+### How to proceed from here
+
+The following steps are needed: 
+1. Make sure consensus regarding the approach is achieved via collaboration on the current document 
+2. Notify all PR (see above) authors about the agreed approach
+3. Implementation: 
+    
+    a. `Plan` is working as expected - either based on #261 above or from scratch. `Plan` should be working correctly regardless of multi-target support
+    
+    b. Extensive testing making sure new `plan` does not introduce any breaking changes
+    
+    c. Change Endpoint struct to support multiple targets - based on #326 - integrate it with new `plan` @sethpollack
+    
+    d. Make sure new endpoint format can still be used in providers which have only partial support for multi targets **TODO**: how ? 
+    
+    e. Add support for multi target which are already addressed in #326. It goes alongside c. and can be based on the same PR @sethpollack. New providers 
+    added since then should maintain same functionality.  
+
+5. Extensive testing on **all** providers before making new release
+6. Update all related documentation and explain how multi targets are supported on per provider basis 
+7. Think of introducing weighted records (see PRs section above) and making them configurable. 
+  
+## Open questions 
+
+- Handling cases when ingress/service targets include both hostnames and IPs
+- "Weighted records scope": https://github.com/kubernetes-incubator/external-dns/issues/196 
```

Comment:
This seem like a problem on top of multi target support which I would keep out of scope. Once we have multi target support it should be fairly simple to extend the ownership model to allow cross cluster records e.g. as outlined in https://github.com/kubernetes-incubator/external-dns/pull/331 or some variation of this.

Reply:
yup